src/os/FileStore.cc

```diff
@@ -641,10 +644,25 @@ bool parse_attrname(char **name)
 
 void FileStore::collect_metadata(map<string,string> *pm)
 {
+  char uuid_str[UUID_LEN+1];
+  blkid_cache cache = NULL;
+  blkid_dev dev = NULL;
+  uuid_d filestore_fsid = get_fsid();
+  
   (*pm)["filestore_backend"] = backend->get_name();
   ostringstream ss;
   ss << "0x" << std::hex << m_fs_type << std::dec;
   (*pm)["filestore_f_type"] = ss.str();
+
+  uuid_unparse((const unsigned char*)&filestore_fsid.uuid, uuid_str);
+
+  if (blkid_get_cache(&cache, NULL) >= 0)
+    dev = blkid_find_dev_with_tag(cache, "PARTUUID", (const char*)uuid_str);
+
+  if (dev)
+    (*pm)["filestore_partition_path"] = string(blkid_dev_devname(dev));
+  else
+    (*pm)["filestore_partition_path"] = "unknown";
```

Comment:
I think that *both* the data and journal partitions will be tagged with the osd UUID, just with different partition types.  We should include both in the metadata?